uarray/_backend.py

```diff
@@ -435,23 +488,36 @@ class Dispatchable:
         Allows one to create a utility function to mark as a given type.
     """
 
-    def __init__(self, value, dispatch_type, coercible=True):
+    def __init__(
+        self,
+        value: _T,
+        dispatch_type: _TT,
+        coercible: bool = True,
+    ) -> None:
         self.value = value
         self.type = dispatch_type
         self.coercible = coercible
 
-    def __getitem__(self, index):
+    @overload
+    def __getitem__(self, index: Literal[0]) -> _TT:
+        ...
+
+    @overload
+    def __getitem__(self, index: Literal[1]) -> _T:
+        ...
+
+    def __getitem__(self, index: Literal[0, 1]) -> _TT | _T:
```

Comment:
I'm not a huge fan of overloads with online annotations, but this is a situation where we really cannot do without.